src/crawler.js

```diff
@@ -71,22 +75,43 @@ function normalizeIpp(ip, port) {
 
 function Crawler(maxRequests, logger) {
   EventEmitter.call(this);
-  this.maxRequests = maxRequests ? maxRequests : 30;
+  
+ 
+  // maxRequests checks
+  maxRequests = maxRequests ? maxRequests : 30;
+  if (!maxRequests || maxRequests < 1) {
+    throw new Error("Invalid max requests");
+  }
+  check.assert.number(maxRequests, "Invalid max requests");
+  
+  // logger checks
+  logger = logger ? logger : console
+  if (!logger.log || !logger.error) {
+    throw new Error("Invalid logger");
+  }
+
+  this.maxRequests = maxRequests;
   this.currentRequests = 0; // active requests
   this.rawResponses = {}; // {$ip_and_port : $normalised_response}
   this.queued = {}; // {$ip_and_port : REQUEST_STATUS.*}
   this.errors = {}; // {$ip_and_port : $error_code_int}
   this.peersData = {}; // {b58_normed(pubKey) : {...}}
-  this.logger = logger || console;
+  this.logger = logger;
 }
 
 util.inherits(Crawler, EventEmitter);
 
 Crawler.prototype.getCrawl = function(entryIp) {
   var self = this;
-  return new Promise(function(resolve, reject){
+  return new Promise(function(resolve, reject) {
+    if (entryIp === undefined) {
+      return reject(new Error("Invalid ip address")) 
+    }
+    if (entryIp.split(' ').length != 1 || !IPP_PATTERN.test(entryIp)) {
+      return reject(new Error("Invalid ip address (perhaps port missing)"))
```

Comment:
You can just ``throw new Error`` without having to worry about calling reject(). It also speeds up execution by a little bit since it doesn't have an extra call to reject().

https://stackoverflow.com/questions/28703241/promise-reject-vs-throw-error

Having an explicit throw in there would also mean you wouldn't need the return statement, and it'd make the code absolutely explicit that it has encountered a failure situation.